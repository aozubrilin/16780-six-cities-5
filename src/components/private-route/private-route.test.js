import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from "react-router-dom";
import history from "../../browser-history";
import PrivateRoute from "./private-route";

const noop = () => {};

describe(`PrivateRoute snapshot`, () => {
  it(`Should PrivateRoute render correctly with authorizationStatus: AUTH`, () => {
    const testStore = configureStore();
    const store = testStore({
      user: {
        authorizationStatus: `AUTH`,
      },
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <PrivateRoute
              exact={true}
              path={`/login`}
              render={noop}
            />
          </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should PrivateRoute render correctly with authorizationStatus: NO_AUTH`, () => {
    const testStore = configureStore();
    const store = testStore({
      user: {
        authorizationStatus: `NO_AUTH`,
      },
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Router history={history}>
            <PrivateRoute
              exact={true}
              path={`/login`}
              render={noop}
            />
          </Router>
        </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
