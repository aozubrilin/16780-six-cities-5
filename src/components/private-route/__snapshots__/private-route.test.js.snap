// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute snapshot Should PrivateRoute render correctly with authorizationStatus: AUTH 1`] = `null`;

exports[`PrivateRoute snapshot Should PrivateRoute render correctly with authorizationStatus: NO_AUTH 1`] = `null`;
